### DISCUSSION TITLE
External pose measurements from MoCap sys

### DISCUSSION THREAD

**Discussion by NageshEranki** [bitcraze/discussions#1057] [Q&A] [answered]
Hi

I am using the motion capture positioning system for my CF. At present, the cflib python library provides a method to send external pose measurements to the CF:
`Crazyflie.extpos.send_extpose()`

1.  What is the highest rate at which I can send these external measurements to the CF over the Crazyradio dongle? More generally speaking, how do I know the frequency of data transfer of any custom data packet I define for that matter?
2. Once I forward the external pose measurements streamed from the cameras to the CF, I understand that it is actually forwarded to the state estimation (EKF) subsystem. Is it possible to bypass the onbard(default) observer and instead access the raw external pose meas, from within the firmware?

Thanks

**Comment by knmcguire** [comment, marked as answer]
Hi!

1. The CRTP protocol from one radio to one crazyflie can handle about 450 packets per second for a payload of 31 bytes. If you add more crazyflies this is shared. This is the port used for localization https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/crtp/crtp_localization/
2. Absolutely. You can observe the 'raw' position before Kalman in this logging variable: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#ext_pos. Do mind that logging takes away some of those packet bandwidth I've previously mentioned :)

**Reply by NageshEranki** [reply]
> The CRTP protocol from one radio to one crazyflie can handle about 450 packets per second for a payload of 31 bytes.

Could you please explain how? The Crazyradio is capable of 2Mbps/1Mbps/250Kbps data-rates. Assuming a one-way transfer rate of 450 packets (32 bytes each) per second, I should be able to transmit and receive at a rate of (450x32x2) bytes/second. 

Which is about 280Kbps. If I set my Crazyradio to use 2Mbps, I should be able to send/receive more packets/second right?

And lastly, could you shed light on how various subsystems share this data-rate? Is it through a port:channel based priority assignment?

Thanks

**Reply by krichardsson** [reply]
Please start a new discussion for the bandwidth question, it is a bit too far from the original title "External pose measurements from MoCap sys"
